src/core/qgis.h

```diff
@@ -1885,6 +1885,26 @@ class CORE_EXPORT Qgis
     };
     Q_ENUM( JoinStyle )
 
+    /**
+     * Flags which control geos geometry creation behavior.
+     *
+     * \since QGIS 3.40
+     */
+    enum class GeosCreationFlag : int SIP_ENUM_BASETYPE( IntFlag )
+    {
+      DontAllowInvalidSubGeom = 1 << 0,  //!< Don't allow invalid sub-geometries to be created
```

Comment:
How about:

```suggestion
      SkipInvalidSubGeometry = 1 << 0,  //!< Don't allow invalid sub-geometries to be created
```

or maybe "Prevent"? "DontAllow" sounds rather clunky to me.

Reply:
It can't be SkipInvalidSubGeometry as it does the exact opposite (i.e., by default code will skip invalid sub geometry, and with this flag it will bail out and returns and empty geom). 

RejectInvalidSubGeometry ?